File: Test/src/main/java/com/example/test/service/HoatDongService.java
package com.example.test.service;

import com.example.test.model.HoatDong;
import com.example.test.model.Statistic;
import com.example.test.payload.request.HoatDongRequest;

import java.util.List;

public interface HoatDongService {
    List<HoatDong> findAll();
    void deleteById(String MaHD);
    HoatDong findById(String MaHD);
    void create(HoatDongRequest hd);
    List<Statistic> statistic();
    void update(HoatDongRequest hd);
}

File: TestFE/src/app/service/hoat-dong.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {HoatDong} from "../model/hoat-dong";
import {Statistic} from "../model/statistic";

const API = 'http://localhost:8080/api/hoatdong';
@Injectable({
  providedIn: 'root'
})
export class HoatDongService {

  constructor(private http: HttpClient) { }

  findAll(): Observable<HoatDong[]> {
    return this.http.get<HoatDong[]>(API + '/findAll');
  }

  findById(id: string): Observable<HoatDong> {
    return this.http.get<HoatDong>(API + '/findById?maHD=' + id);
  }

  delete(maHD: string): Observable<any> {
    return this.http.get<any>(API + '/delete?maHD=' + maHD);
  }

  create(hoatDong): Observable<any> {
    return this.http.post<any>(API + '/create', hoatDong);
  }

  statistic(): Observable<Statistic[]> {
    return this.http.get<Statistic[]>(API + '/statistic');
  }

  update(hoatdong): Observable<any> {
    return this.http.post<any>(API + '/update', hoatdong);
  }
}
